### PR TITLE
Refactor Resource Form: Create new broadcast upon successful resource form submission

### PR DESCRIPTION
**Pull request by shivamgaur99**
Closes: #1021 

##Description

Refactor the resource form so that a new broadcast gets created on successful submission

## Screenshots
image
image
image

_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (Documentation content changed)
- [ ] Other (please describe): 

## Checklist

_Put an `x` in the boxes that apply_

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] My changes does not break the current system and it passes all the current test cases.

## Other information

Any other information that is important to this pull request

### CODE REVIEW

**Comment by shivamgaur99** [comment]
@Kajol-Kumari Please review and let me know if any changes are required.

File: backend/app/routes/resources/@validationSchema/index.js [hunk 1]
```diff
@@ -1,13 +1,27 @@
...
 
-const ResourcesValidationSchema = Joi.object().keys({
-  name: Joi.string().trim().required().min(3),
-  email: Joi.string().trim().email().required(),
-  url: Joi.string().trim().required().min(10),
-  description: Joi.string().trim().required().min(5),
-  trustLevel: Joi.number().integer().required(),
-  expiryDate: Joi.date().required(),
-  additionalInfo: Joi.string().trim().min(5),
+// const ResourcesValidationSchema = Joi.object().keys({
```
Comment (Kajol-Kumari) [inline backend/app/routes/resources/@validationSchema/index.js:3]: please remove the commented code as it won't be needed anymore

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
Screenshot 2024-06-08 at 12 10 24 PM
I am getting this error while trying to create a resource, can you fix it?

File: backend/app/routes/resources/@validationSchema/index.js (hunk 1 above)
Comment (shivamgaur99) [reply backend/app/routes/resources/@validationSchema/index.js:3]: okay, also, we don't need resource model as we are using brodcast model and previous backend code addresource, get resource , deleteResource: should i remove it or skip that?

**Comment by shivamgaur99** [comment]
> Screenshot 2024-06-08 at 12 10 24 PM I am getting this error while trying to create a r

> Screenshot 2024-06-08 at 12 10 24 PM I am getting this error while trying to create a resource, can you fix it?


@Kajol-Kumari The error was occurring for me too when I made changes in the backend code, but it got fixed when I restarted my application.

**Comment by shivamgaur99** [comment]
@Kajol-Kumari i configured .env and tested in my local 
image

File: backend/app/routes/resources/@validationSchema/index.js (hunk 1 above)
Comment (Kajol-Kumari) [reply backend/app/routes/resources/@validationSchema/index.js]: Please remove it as it will just redundant code after this change

Comment (shivamgaur99) [reply backend/app/routes/resources/@validationSchema/index.js]: image


also removing it from the admin dashboard as we are managing all resources in admin dashboard

Comment (Kajol-Kumari) [reply backend/app/routes/resources/@validationSchema/index.js]: yeah, that should be fine

**Review by Kajol-Kumari — APPROVED** [review]
LGTM

**Comment by Kajol-Kumari** [comment]
@shivamgaur99 one thing i forgot, where is the `isApproved` flag? as soon as we save the resource, the broadcast should get created with the flag set to false by default. Once the admins approve it, it should set to true

**Comment by Kajol-Kumari** [comment]
@shivamgaur99 please make sure to introduce the `isApproved` flag in broadcast schema having default value set to `false`

**Comment by Kajol-Kumari** [comment]
@shivamgaur99 please let me know when the pr becomes ready for review(isApproved flag is yet missing)

**Comment by shivamgaur99** [comment]
i will make pr by morning after testing everything is working as expected

**Comment by shivamgaur99** [comment]
Please review, all changes have been made :)

File: backend/index.js [hunk 2]
```diff
@@ -8,47 +8,12 @@ const cluster = require('./helpers/cluster');
...
-const Resource = require('./app/models/resource');
 const {
   JoinUsCronJobMailTemplate,
   ContactUsCronJobMailTemplate,
-  ResourceDeletedMailTemplate,
 } = require('./utility/emailTemplates');
 const Admin = require('./app/models/Admin');
 
-// Cron job to delete all resource data for 2months
-cronJob('0 0 2 * *', async (req, res, next) => {
```
Comment (Kajol-Kumari) [inline backend/index.js:20]: we might need the cron job later to delete the broadcast/contactUs data which are older than 6 months/1 year as they will just keep on accumulating no?, for now it should be fine

**Review by Kajol-Kumari — APPROVED** [review]

File: backend/index.js (hunk 2 above)
Comment (shivamgaur99) [reply backend/index.js:20]: Since there was no cron job for the broadcast, I didn't add it. Can I add it to the current PR?

**Pull request merged**